rofs/downloader.go

```diff
@@ -46,7 +48,26 @@ func (d *Downloader) DownloadBlock(block meta.BlockInfo) ([]byte, error) {
 		return nil, err
 	}
 
-	return snappy.Decode(nil, xxtea.Decrypt(data, block.Decipher))
+	data, err = snappy.Decode(nil, xxtea.Decrypt(data, block.Decipher))
+	if err != nil {
+		return nil, err
+	}
+
+	hasher, err := blake2b.New(16, nil)
```

Comment:
Do you need a new hasher everytime ? can't we have a single hasher and just reset it? Would avoid allocation